static/panes/compiler.js

```diff
@@ -321,7 +350,72 @@ Compiler.prototype.resize = function () {
     });
 };
 
+// Returns a label name if it can be found in the given position, otherwise
+// returns null.
+Compiler.prototype.getLabelAtPosition = function (position) {
+    var line = this.outputEditor.getModel().getLineContent(position.lineNumber);
```

Comment:
with annotated asm I think this would be as "simple" as `return this.assembly[line-1].label` (once we define `label` in the server-side)